### ISSUE TITLE
QR Code scanner now just displays black screen after updating to v23

### ISSUE THREAD

**Issue by Dygerati** [expo/expo#975]
This is actually true of both the updated Expo app, as well as my own -- I can no longer scan QR codes. It just displays a black screen. I know that the scanner was changed to utilize the camera API in this version so it's likely related to that. 

Is anyone else seeing this issue? I've only been able to test on Android dev so far, but I'll be testing on iOS shortly. 

I'm using a Oneplus 3T running Oxygen `4.5.0`

**Comment by Dygerati** [comment]
I noticed the most recent review on the Play Store mentions the same issue, so it seems it's not just me at least:

Via Rakesh R:
>Once the camera is opened to scan QR code The screen just turns black Doesn't even open the camera This happened on the latest update sdk 23

**Comment by Dygerati** [comment]
Oh, and I confirmed that the issue is *not* occurring on my iPhone 5s.

**Comment by jesseruder** [comment]
Thanks for the info. We'll look into it.

**Comment by Dygerati** [comment]
Sounds like this is the same issue, recorded in the forums:
https://forums.expo.io/t/black-screen-for-camera/3587/11

The discussion is from early Oct., and it seems that the issue is now impacting the BarCodeScanner as well since it's now using the camera component.

**Comment by reggie3** [comment]
As another data point, I am also experiencing the black screen when using the camera on a OnePlus device (I am using a One Plus 5).  The problem occurs in my own application.  I also occurs in the updated camerja community example.

**Comment by sameesiddiqui** [comment]
I have a similar issue: 
barcode scanner works when i open it for the first time and open an app, but if i try to open the bar code scanner after opening an app before, it just shows a black screen (and a visual glitch at the bottom of the screen with part of the barcode scanner screen)

this can get fixed by leaving the expo app (don't even need to kill it), then reopening it and opening the bar code scanner again. the process then repeats, where i can open 1 app and it will not work unless i do the same dance.

This is on a samsung galaxy s8+ after updating to expo 23.
This issue does not occur on iphone 6 that I tested with

**Comment by mmazzarolo** [comment]
I'm having the same issue using the barcodeScanner.  
Black camera screen and hard crash after updating to expo 23. Tried with both `BarcodeScanner` and `Camera`.  
It was working fine before updating. 

```
11-21 10:49:20.076  7580  7596 E AndroidRuntime: Process: host.exp.exponent, PID: 7580
11-21 10:49:20.076  7580  7596 E AndroidRuntime:        at abi23_0_0.host.exp.exponent.modules.api.components.camera.ExpoCameraView$ReaderAsyncTask.doInBackground(ExpoCameraView.java:379)
11-21 10:49:20.076  7580  7596 E AndroidRuntime:        at abi23_0_0.host.exp.exponent.modules.api.components.camera.ExpoCameraView$ReaderAsyncTask.doInBackground(ExpoCameraView.java:356)
```

*Device:* Asus Tablet P00C - 6.0.0 - API 23

**Comment by fkuster** [comment]
Hi, I’ve also noticed same issue with barcode scanner component after updating to v23.  Will this be fixed soon? Any suggestions?

**Comment by brentvatne** [comment]
we are working on a fix right now, apologies!

**Comment by fkuster** [comment]
Barcode scanner works for me now on Samsung S8 and S5 but smartphone has to be oriented vertically(on iphone works horizontally also) and component should be unmounted when back button is clicked.

**Comment by ashconnell** [comment]
I'm also having this issue. `Camera` will mount and unmount reliably over and over again - but as soon as I call `ImagePicker.launchImageLibraryAsync` then `Camera` will just be blank and trigger `onMountError`.

**Comment by aalices** [comment]
An update was released for Android (https://blog.expo.io/expo-for-android-sdk23-minor-update-daf5847ae0a7) and should fix the black screen issue, hope it helps

**Comment by Dygerati** [comment]
I just rebuilt and tested the standalone on my Android device, and the BarCodeScanner still just displays a black screen on my Oneplus 3T.

**Comment by ashconnell** [comment]
Yep I rebuilt my project with the latest sdk and opening the library still breaks the camera component

**Comment by fkuster** [comment]
I switched to v22 until this is solved and it works normally.

**Comment by quentinor** [comment]
Hi, I am experiencing the same kind of issues with my oneplus2, trying to access the camera module.

Can someone tell me if the android update (https://blog.expo.io/expo-for-android-sdk23-minor-update-daf5847ae0a7) is supposed to fix this problem (didn't seem for me), or this behavior can be reproduced wit other devices from other brands ?

I tried going back to version 22 @fkuster , but it didn't seem to work either. Just to be sure, I ran my app on a Nexus 5X, A Neffos YL, a Samsung Tab 4 active ... and it worked just fine :)

If someone found a workaround, or has any information about the fix/upgrade, that would be much appreciated!

**Comment by fkuster** [comment]
@quentinor When you bumped version to 22, did you load app through expo or you made apk? If you loaded app through expo mobile app, expo servers will return newest code in neweset expo version, and this is 23 in your case. (I'm not sure about that but i think i saw this somewhere)
I built  apk in version 22 and then it was ok, but through expo it wasn't.

**Comment by aalices** [comment]
@quentinor we had issues with OnePlus devices but they should be fixed with SDK 24 that releases very soon!

**Comment by EladAvni** [comment]
I'm having the same issues with a Galaxy J7 - updated to Android 7.0 using update J710FXXU3BQJ4 and both the camera on my app and the barcode scanner on the Expo client stopped working.

Any news on the update?

**Comment by rebitte** [comment]
Any news on the update?

**Comment by lmichailian** [comment]
Same problem for samsung galaxy S8 :/

**Comment by fkuster** [comment]
I'm using barcode scanner module in production. Everything works fine in expo v22. I think it is still not fixed in v23 and v24.

**Comment by aalices** [comment]
Sorry for any inconvenience with all the Scanner/Camera stuff, it's difficult to keep track because it happens on devices I haven't got. Any Logcat output would be appreciated!

I'm working on exposing Camera `start`/`stop` and hoping it will be a kind of cure!

**Comment by rebitte** [comment]
Hi Guys, I would like to hire the best developer in Expo.io to solve my problem with Black Screen.

**Comment by lmichailian** [comment]
I resolved this doing a downgrade to v22 like @fkuster says

**Comment by rebitte** [comment]
I can't downgrade to v22 :(

**Comment by deepkolos** [comment]
i have this problem too :(

**Comment by Dygerati** [comment]
This actually now works on my Oneplus devices on v24--was it not fixed in this version for others?

**Comment by mmazzarolo** [comment]
I'm on v25 and it seems to be working fine now!

**Comment by lmichailian** [comment]
Not for me :/ @mmazzarolo  on wich device works for you?

**Comment by mmazzarolo** [comment]
Hey @lmichailian , I tried it on an `Asus Tablet P00C - 6.0.0 - API 23` that had the issue prior v25.  

For anyone interested, here is a small list of tips I drafted while trying to make the Camera work:
* After the expo upgrade (which should be executed by following the upgrade guide) clear all the expo cache from the device (I uninstalled it) and restart the packager with a reset (`npm run start -- --reset-cache`)
* Make sure you're using the `Camera` component instead of the old `BarcodeScanner` component
* If you're using the barcode scanner functionality (by using the `onBarCodeRead` prop) make sure that you're rendering the component only if the `Permissions.CAMERA` has been granted, otherwise the app will just crash/show a black screen

**Comment by timhuff** [comment]
I have a workaround that might interest future travelers of this thread - figured I'd throw it out there. After spending 5 hours trying to solve the problem directly, I came up with the idea to use `WebBrowser.openSessionAsync` to open a browser, direct them to `https://site.com/mobileAppUpload?token=${token}`, and populate that page with dressed up input elements of the form `<input type="file" accept="image/*" capture="camera" />`. When these are clicked, the camera opens and the user is prompted to take a photo.

After running through a proof of concept, the user experience isn't terrible, depending on your audience (luckily, my app is a tool for a company to document things so I doubt the employees will mind much). It's not too different from a modal. And users are used to these "child browsers" for the purpose of authentication and such.

**Comment by mmazzarolo** [comment]
Thanks @timhuff for the quite interesting workaround

**Comment by MadeoIT** [comment]
I am having this issue also

**Comment by mmazzarolo** [comment]
Hey @aalices , why has the isse been closed? 🤔

**Comment by aalices** [comment]
This was caused by BarCodeScanner being based on Camera and it's not anymore. Feel free to report any problems and we could reopen this

**Comment by rusiyo** [comment]
I'm still seeing this issue on Android using "expo": "^27.0.1"

**Comment by aalices** [comment]
@rusiyo can you create a snack reproducing your case?
